src/wallet/rpcwallet.cpp

```diff
@@ -1229,9 +1227,8 @@ UniValue ListReceived(const UniValue& params, bool fByAccounts)
         if (wtx.IsCoinBase() || !CheckFinalTx(wtx))
             continue;
 
-        int nDepth = wtx.GetDepthInMainChain();
         int nBCDepth = wtx.GetDepthInMainChain(false);
-        if (nDepth < nMinDepth)
+        if (wtx.GetDepthInMainChain() < nMinDepth)
```

Comment:
Shouldn't this be `if (nBCDepth < nMinDepth)` ? Otherwise it feels like we are taking IS "confirmations" into account which could be confusing.